lib/inc/hocon/config_mergeable.hpp

```diff
@@ -6,6 +6,7 @@
 namespace hocon {
 
     class LIBCPP_HOCON_EXPORT config_mergeable {
+        friend class config_value;
```

Comment:
http://stackoverflow.com/a/4672888 answers the friend class requirement. We can revisit later if we'd like to just make some methods public instead of using friend everywhere.